test/test_plugin_twitter.py

```diff
@@ -419,161 +501,151 @@ def test_plugin_twitter_edge_cases():
-    bad_media_response.content = dumps({
-        "errors": [
-            {
-                "code": 93,
-                "message": "This application is not allowed to access or "
-                "delete your direct messages.",
-            }]})
-    bad_media_response.status_code = requests.codes.internal_server_error
+    mock_get = mocker.patch("requests.get")
+    mock_post = mocker.patch("requests.post")
 
-    mock_post.side_effect = [good_media_response, good_dm_response]
-    mock_get.return_value = good_dm_response
+    mock_post.side_effect = [good_media_response, good_message_response]
+    mock_get.return_value = good_message_response
 
-    twitter_url = 'twitter://{}/{}/{}/{}'.format(ckey, csecret, akey, asecret)
-
-    # attach our content
-    attach = AppriseAttachment(os.path.join(TEST_VAR_DIR, 'apprise-test.gif'))
-
-    # instantiate our object
+    # Create application objects.
     obj = Apprise.instantiate(twitter_url)
+    attach = AppriseAttachment(os.path.join(TEST_VAR_DIR, 'apprise-test.gif'))
 
     # Send our notification
     assert obj.notify(
         body='body', title='title', notify_type=NotifyType.INFO,
         attach=attach) is True
 
-    # Test our call count
+    # Test call counts.
+
+    # Calls to `verify_credentials.json` will get cached by `NotifyTwitter`.
+    # TODO: Review this.
     assert mock_get.call_count == 1
     assert mock_get.call_args_list[0][0][0] == \
         'https://api.twitter.com/1.1/account/verify_credentials.json'
+
     assert mock_post.call_count == 2
     assert mock_post.call_args_list[0][0][0] == \
         'https://upload.twitter.com/1.1/media/upload.json'
     assert mock_post.call_args_list[1][0][0] == \
         'https://api.twitter.com/1.1/direct_messages/events/new.json'
 
-    mock_get.reset_mock()
-    mock_post.reset_mock()
 
-    # Test case where upload fails
-    mock_get.return_value = good_dm_response
-    mock_post.side_effect = [bad_media_response, good_dm_response]
+def test_plugin_twitter_dm_attachments_bad_media_response(
+        mocker, twitter_url,
+        good_message_response, good_media_response, bad_media_response):
+    """
+    Test case with a bad media response.
+    """
 
-    # instantiate our object
+    mock_get = mocker.patch("requests.get")
+    mock_post = mocker.patch("requests.post")
+
+    mock_get.return_value = good_message_response
+    mock_post.side_effect = [good_media_response, bad_media_response]
+
+    # Create application objects.
     obj = Apprise.instantiate(twitter_url)
+    attach = AppriseAttachment(os.path.join(TEST_VAR_DIR, 'apprise-test.gif'))
 
     # Send our notification; it will fail because of the media response
     assert obj.notify(
         body='body', title='title', notify_type=NotifyType.INFO,
         attach=attach) is False
 
-    # Test our call count
+    # No get request as cached response is used.
+    # TODO: Review this.
     assert mock_get.call_count == 0
```

Comment:
This will fail when being invoked with `pytest test/test_plugin_twitter.py -nauto`.

Reply:
Fixed.